File: 프로그래머스/lv1/42748. K번째수/K번째수.kt
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        val arr = IntArray(commands.size) { 0 }

        for(i in commands.indices) {
            val firstNumber = commands[i][0] - 1
            val lastNumber = commands[i][1] - 1
            val sortingNumber = commands[i][2] - 1

            arr[i] = array.filterIndexed { index, i -> index in firstNumber..lastNumber }.sorted()[sortingNumber]
        }

        return arr
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.swift
func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
    var answer: [Int] = []
    
    for c in commands {
        let firstCondition = array[c[0]-1...c[1]-1]
        let secondCondition = firstCondition.sorted()
        let thirdCondition = secondCondition[c[2]-1]
        
        answer.append(thirdCondition)
    }
    
    return answer
}